cmd/nerdctl/container_create.go

```diff
@@ -50,36 +51,433 @@ func newCreateCommand() *cobra.Command {
 	return createCommand
 }
 
-func createAction(cmd *cobra.Command, args []string) error {
-	globalOptions, err := processRootCmdFlags(cmd)
+func processContainerCreateOptions(cmd *cobra.Command) (opt types.ContainerCreateOptions, err error) {
+	opt = types.ContainerCreateOptions{}
+	opt.Stdout = cmd.OutOrStdout()
+	opt.Stderr = cmd.ErrOrStderr()
+	opt.GOptions, err = processRootCmdFlags(cmd)
 	if err != nil {
-		return err
+		return
 	}
-	platform, err := cmd.Flags().GetString("platform")
+
+	opt.NerdctlCmd, opt.NerdctlArgs = globalFlags(cmd)
+
+	// #region for basic flags
+	opt.Interactive, err = cmd.Flags().GetBool("interactive")
 	if err != nil {
-		return err
+		return
+	}
+	opt.TTY, err = cmd.Flags().GetBool("tty")
+	if err != nil {
+		return
+	}
+	opt.Restart, err = cmd.Flags().GetString("restart")
+	if err != nil {
+		return
+	}
+	opt.Rm, err = cmd.Flags().GetBool("rm")
+	if err != nil {
+		return
+	}
+	opt.Pull, err = cmd.Flags().GetString("pull")
+	if err != nil {
+		return
+	}
+	opt.Pid, err = cmd.Flags().GetString("pid")
+	if err != nil {
+		return
+	}
+	opt.UTS, err = cmd.Flags().GetString("uts")
+	if err != nil {
+		return
+	}
+	opt.StopSignal, err = cmd.Flags().GetString("stop-signal")
+	if err != nil {
+		return
+	}
+	opt.StopTimeout, err = cmd.Flags().GetInt("stop-timeout")
+	if err != nil {
+		return
+	}
+	// #endregion
+
+	// #region for platform flags
+	opt.Platform, err = cmd.Flags().GetString("platform")
+	if err != nil {
+		return
+	}
+	// #endregion
+
+	// #region for init process flags
+	opt.InitProcessFlag, err = cmd.Flags().GetBool("init")
+	if err != nil {
+		return
+	}
+	opt.InitBinaryChanged = cmd.Flags().Changed("init-binary")
+	opt.InitBinary, err = cmd.Flags().GetString("init-binary")
+	if err != nil {
+		return
+	}
```

Comment:
can we use a string pointer (InitBinary *string)? so we know `cmd.Flags().Changed("init-binary")` is true if the pointer is not nil.

Reply:
same for other flags where `Changed` needs to be checked (opt.Network, opt.Net, opt.Publish, opt.DNS)

Reply:
I think it's okay to have a default value, as it doesn't affect the logic judgment here.